asyncscope.md

```diff
@@ -530,8 +609,30 @@ struct async_scope {
 
 2. The destructor will call `terminate()` if there is outstanding work in the `async_scope` object (i.e., work created by `spawn()` and `spawn_future()` did not complete).
 
-3. *Note*: It is always safe to call the destructor after the sender returned by `empty()` sent the completion signal, provided that there were no calls to `spawn()` and `spawn_future()` since `empty()` was called, or stop was requested on our stop source before any of these `spawn()` and `spawn_future()` calls.
+3. *Note*: It is always safe to call the destructor after the sender returned by `on_empty()` sent the completion signal, provided that there were no calls to `spawn()` and `spawn_future()` since `on_empty()` was called, or stop was requested on our stop source before any of these `spawn()` and `spawn_future()` calls.
+
+## `async_scope::nest`
+
+1. `async_scope::nest` is used to produce a `nested_sender` that nests the sender within the lifetime of the `async_scope` object. The `nested_sender` will start the given sender when it is started.
 
+2. The returned sender has the same completion signatures as the input sender.
+
+3. *Effects*:
+  - The `async_scope` will not be `empty` 
+    - until the returned `nested_sender` is destroyed 
+    - or the returned `nested_sender` is started and the operation completes
```

Comment:
Why do we keep `async_scope` non-empty while the returned sender is still active? Is this to avoid allocations?

It would be great if `nest` could simply replace `spawn_future`.

The thing that I'm afraid with this wording is extending the lifetime of nested_scope for too long. Example:
```c++
async_scope s;

sender auto res = s.nest(just() | then(&work_to_keep_in_scope)) | then(&very_long_operation);
// keep track of lifetime of res

s.set_stopped();
std::this_thread::sync_wait(s.on_empty());
```

Reply:
nest() only extends the scope, it does not eagerly start the given sender. This is why it does not need to allocate and must hold the scope until the returned sender is discarded or started and completed.